File: backend/app/schemas/resume.py
"""Request/response shapes for the resume module (Phase 5). `raw_text` is
intentionally excluded from the public schema — it can be several KB and
the frontend only ever needs `parsed_data`'s summarized extraction plus the
AI fields.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.resume import ResumeStatus


class ResumePublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    original_filename: str
    mime_type: str
    file_size_bytes: int
    status: ResumeStatus
    parse_error: str | None
    parsed_data: dict | None
    ai_summary: str | None
    ai_strengths: list[str] | None
    ai_suggestions: list[str] | None
    ai_score: int | None
    ai_provider_used: str | None
    created_at: datetime
    updated_at: datetime


class ResumeResponse(BaseModel):
    data: ResumePublic


class ResumeListResponse(BaseModel):
    data: list[ResumePublic]

File: frontend/types/resume.ts
export type ResumeStatus = "uploaded" | "parsing" | "parsed" | "failed";

export interface ResumeParsedData {
  emails: string[];
  phones: string[];
  urls: string[];
  github_url: string | null;
  linkedin_url: string | null;
  detected_skills: string[];
}

export interface Resume {
  id: string;
  original_filename: string;
  mime_type: string;
  file_size_bytes: number;
  status: ResumeStatus;
  parse_error: string | null;
  parsed_data: ResumeParsedData | null;
  ai_summary: string | null;
  ai_strengths: string[] | null;
  ai_suggestions: string[] | null;
  ai_score: number | null;
  ai_provider_used: string | null;
  created_at: string;
  updated_at: string;
}
